Jenkinsfile

```diff
@@ -0,0 +1,16 @@
+node {
+    stage 'Checkout'
+    checkout scm
+
+    stage 'Build virtualenv'
+    sh '''virtualenv venv
+    . ./venv/bin/activate
+    pip install --upgrade pip --no-cache-dir
+    pip install -e .
+    pip install -r tests/requirements.txt'''
+
+    stage 'Tests'
+    sh '''. venv/bin/activate
+    nosetests --with-xcoverage --with-xunit'''
```

Comment:
Tox is unrelated to that, you would run nose as part of a tox test environment. It's more about the experience I've made with nosetest and py.test over time: they do a lot of magic. When they work they're great. When tests blow up you might have a hard time to debug what happened.